src/relax/transform/decompose_composite_ops.cc

```diff
@@ -110,21 +111,60 @@ class NormInferenceSimplifier : public ExprMutator {
   Map<Expr, Expr> batch_norm_map_;
 };
 
+class OpDecomposer : public ExprMutator {
+ public:
+  static Expr Decompose(Expr expr) { return OpDecomposer()(expr); }
+
+ private:
+  using ExprMutator::VisitExpr_;
+  Expr TensorToShape(const Call& call_node) {
+    ICHECK(call_node->args.size() == 1);
```

Comment:
seems the ICHECK is duplicate, the check in ReturnTensorToShapeStructInfo should be good to cover this